### PR TITLE
Add Windows MSYS2 Tests to pull-request.yml

### PR DESCRIPTION
**Pull request by WardenGnaw**
Add MSYS2 tests to Windows build to enable GNU debugging testing.

### CODE REVIEW

File: test/CppTests/TestConfigurations/config_msys_gdb.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <TestConfiguration
+      Compiler="GppMingw64"
+      DebuggeeArchitecture="x64"
+      Debugger="GdbMingw64" />
+  </TestConfigurations>
+  <Compilers>
+    <Compiler
+      Name="GppMingw64"
+      Type="GPlusPlus"
+      Path="D:\a\_temp\msys\msys64\mingw64\bin\g++.exe"/>
```
Comment (gregg-miskelly) [inline test/CppTests/TestConfigurations/config_msys_gdb.xml:13]: Is it okay that these are hard coded?

**Review by gregg-miskelly — APPROVED** [review]
Is there something we can do to suppress the notification for all the compiler warnings? Or do we just need to prioritize disabling or fixing them?

Otherwise LGTM.

File: test/CppTests/TestConfigurations/config_msys_gdb.xml (hunk 1 above)
Comment (WardenGnaw) [reply test/CppTests/TestConfigurations/config_msys_gdb.xml:13]: > Is there something we can do to suppress the notification for all the compiler warnings? Or do we just need to prioritize disabling or fixing them?

Working on resolving the warnings right now. 

> Is it okay that these are hard coded?

I think for now that is ok. I want to have at least Windows/Linux tests running so we can review other Contributors PR's.

I also want to enable Cygwin/MinGW as separate tasks along side MSYS.
- https://github.com/marketplace/actions/install-cygwin
- https://github.com/marketplace/actions/install-mingw

Comment (WardenGnaw) [reply test/CppTests/TestConfigurations/config_msys_gdb.xml:13]: Ideally we run in msys `cygpath -w $(which [g++|gdb])` for that transform.

**Pull request merged**